packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -147,23 +104,38 @@ library LibProving {
         );
 
         if (fcId == 0) {
-            if (config.enableOracleProver) revert L1_NOT_ORACLE_PROVEN();
-            fcId = _getNextForkChoiceId(blk);
+            fcId = blk.nextForkChoiceId;
+
+            unchecked {
+                ++blk.nextForkChoiceId;
+            }
+
+            fc = blk.forkChoices[fcId];
+
+            if (fcId == 1) {
+                // We only write the key when fcId is 1.
+                fc.key = LibUtils.keyForForkChoice(
```

Comment:
looks like this `fc.key` may not work as expected since right now we allow that the first prover is not an oracle prover?

Reply:
`fc.key `has nothing to do with oracle prover.  It's just a trick to reduce gas. 

Now the differences between oracle prover a regular prover are:
- the oracle prover can override existing proofs as long as the block is not verified.
- after a oracle proof, a regular proof must follow before the block can be verified.